### PR TITLE
test: put conv in one reduce

### PR DESCRIPTION
**Pull request by geohot**
This should be in the scheduler, not lazy.

Also that ratio thing brings up an interesting issue.

### CODE REVIEW

**Comment by geohot** [comment]
`SPLIT_REDUCEOP=0 DEBUG=2 BEAM=2 python3 examples/beautiful_mnist.py`

**Comment by geohot** [comment]
```
ShapeTracker(views=(View(shape=(512, 32, 5, 25, 5, 25, 32), strides=(0, 25, 5, 0, 1, 0, 800), offset=0, mask=((0, 512), (0, 32), (0, 5), (0, 20), (0, 5), (0, 20), (0, 32)), contiguous=False), View(shape=(512, 32, 144, 144, 32), strides=(16000000, 500000, 4000, 32, 1), offset=0, mask=((0, 5
12), (0, 32), (0, 125), (0, 125), (0, 32)), contiguous=False), View(shape=(64, 2, 24, 3, 16, 8, 6, 6, 8, 4, 2, 4), strides=(169869312, 10616832, 4608, 256, 663552, 32, 110592, 768, 4, 21233664, 84934656, 1), offset=0, mask=None, contiguous=False)))
ShapeTracker(views=(View(shape=(512, 32, 5, 25, 5, 25, 32), strides=(12800, 0, 0, 20, 0, 1, 400), offset=0, mask=((0, 512), (0, 32), (0, 5), (0, 20), (0, 5), (0, 20), (0, 32)), contiguous=False), View(shape=(512, 32, 144, 144, 32), strides=(16000000, 500000, 4000, 32, 1), offset=0, mask=((0, 512), (0, 32), (0, 125), (0, 125), (0, 32)), contiguous=False), View(shape=(64, 2, 24, 3, 16, 8, 6, 6, 8, 4, 2, 4), strides=(169869312, 10616832, 4608, 256, 663552, 32, 110592, 768, 4, 21233664, 84934656, 1), offset=0, mask=None, contiguous=False)))
```

**Comment by geohot** [comment]
Two equivalent ShapeTrackers:
```
View(shape=(1, 1, 2, 4, 2, 4), strides=(0, 0, 2, 8, 1, 4), offset=0, mask=((0, 1), (0, 1), (0, 2), (0, 2), (0, 2), (0, 2)), contiguous=False)
View(shape=(1, 1, 9, 9), strides=(0, 0, 8, 1), offset=0, mask=((0, 1), (0, 1), (0, 8), (0, 8)), contiguous=False)
View(shape=(1, 1, 1, 1, 3, 3, 3, 3), strides=(0, 0, 0, 0, 27, 9, 3, 1), offset=0, mask=None, contiguous=True)
```

```
View(shape=(1, 1, 2, 4, 2, 4), strides=(0, 0, 2, 8, 1, 4), offset=0, mask=((0, 1), (0, 1), (0, 2), (0, 2), (0, 2), (0, 2)), contiguous=False)
View(shape=(1, 1, 1, 1, 3, 3, 3, 3), strides=(0, 0, 0, 0, 24, 8, 3, 1), offset=0, mask=((0, 1), (0, 1), (0, 1), (0, 1), (0, 2), (0, 3), (0, 2), (0, 3)), contiguous=False)
```

@corsix You interested in improving simplify to handle this?

**Comment by Qazalin** [comment]
Moved everything to the scheduler, TestSchedule.test_fold_conv_relu_backward is green.

Tomorrow:

- the reshapes and permutes don't generalize to the arange fusion. Is there some sort of generic ShapeTracker witchcraft that'd handle all reduce -> expand -> reduce patterns. Push the expand through, then change the axes and reshape on both reduces. 
- can I always fold (ReduceOps.SUM, src=(ReduceOps.SUM,))? That way this doesn't get in the way of arange fusion.
- restore the test_fold_conv_relu_backward_half behavior

File: test/test_arange.py [hunk 1]
```diff
@@ -67,6 +69,7 @@ def test_index(self):
       #assert GlobalCounters.global_ops < 4*16384, f"too many ops {GlobalCounters.global_ops}"
     np.testing.assert_allclose(real_index, X.numpy())
 
+  @unittest.skip("TODO: fix this")
```
Comment (Qazalin) [inline test/test_arange.py:72]: I'm doing this as a part of https://github.com/tinygrad/tinygrad/pull/5590

**Pull request merged**